flow/display_list_canvas.cc

```diff
@@ -20,7 +33,7 @@ void DisplayListCanvasDispatcher::restore() {
 void DisplayListCanvasDispatcher::saveLayer(const SkRect* bounds,
                                             bool restore_with_paint) {
   TRACE_EVENT0("flutter", "Canvas::saveLayer");
-  canvas_->saveLayer(bounds, restore_with_paint ? &paint() : nullptr);
+  canvas_->saveLayer(bounds, safe_paint(restore_with_paint));
```

Comment:
I've already realized that this particular line needs to push/restore the extra alpha otherwise the contents of the saveLayer will have the extra_alpha applied as will the restore - doubling the effect. I'll be fixing that soon.

Reply:
Done...